### DISCUSSION TITLE
How do I get colors on the mode with mini.statusline

### DISCUSSION THREAD

**Discussion by a73s** [nvim-mini/mini.nvim#1037] [Q&A] [answered]
It says it supports colors but i cant figure it out.
here is my config for lazy nvim
```
        {
		'echasnovski/mini.nvim',
		config = function()
			
			local statusline = require 'mini.statusline'
			statusline.setup({use_icons = vim.g.have_nerd_font})

			---@diagnostic disable-next-line: duplicate-set-field
			statusline.section_location = function()
				return '%p%%, %.l:%2L:%.c'
			end

		end,
	},
```

**Comment by echasnovski** [comment, marked as answer]
They are defined with `MiniStatuslineModeXxx` highlight groups. Here is a [full list](https://github.com/echasnovski/mini.nvim/blob/072ef026a538cf4a9dedba56ad8eecffa9e9abb6/doc/mini-statusline.txt#L51-L57). By default they are linked to some built-in highlight groups, which can look not as pretty as you'd expect.

Either use a color scheme that defines them explicitly (for example, execute `:colorscheme randomhue`) or define them manually (see `:h nvim_set_hl()` and for example of the logic - [how it is done in 'mini.hues'](https://github.com/echasnovski/mini.nvim/blob/072ef026a538cf4a9dedba56ad8eecffa9e9abb6/lua/mini/hues.lua#L1162-L1167)).
